utils/misc.mli

```diff
@@ -186,6 +186,10 @@ val find_in_path_uncap: string list -> string -> string
            to match. *)
 val remove_file: string -> unit
         (* Delete the given file if it exists. Never raise an error. *)
+val remove_regular_file: string -> unit
+        (* Delete the given file if it exists and is a regular file.
+           Do nothing for other kinds of files.
+           Never raise an error. *)
```

Comment:
Note: instead of adding a new function, we could also change `Misc.remove_file` so that it does nothing when the file is not a regular file.  It would be just as efficient as today and perhaps less error-prone overall.  Opinions welcome.

Reply:
Looking at the other uses of `Misc.remove_file` in the distribution, there appears to be no valid reason to ever remove a regular file.  All the other uses are either a tmpfile, or a user-specified predictable output (like a cmofile).

So just making `Misc.remove_file` skip a non-regular file deletion seems simplest to me.

Reply:
I think it's better to stop adding ad-hoc file system calls to `Sys`. People should simply use `Unix` if they are serious about interacting with file systems.

Reply:
(In particular these ad-hoc file system calls tend to report disputable results on permission errors, which leads to bad usability and head scratching, e.g. `Sys.file_exists` returning `false` on permission errors).

Reply:
I don't like adding syscalls to `Sys` either, but we're definitely not ready to make the compiler (and the whole bootstrap) depend on the Unix library.